scripts/retrieve_databundle_light.py

```diff
@@ -496,6 +497,9 @@ def get_best_bundles_by_category(
     if dict_n_matched:
         # if non-empty, then pick bundles until all countries are selected
         # or no more bundles are found
+        df2 = pd.DataFrame()
+        df2["number of matches"] = dict_n_matched
+        df = pd.concat([df, df2])
         dict_sort = sorted(dict_n_matched.items(), key=lambda d: d[1])
 
         current_matched_countries = []
```

Comment:
I confirm that this place is where the dataframe shall be placed.
You may initialize the empty dataframe as:
`df_matched = pd.DataFrame(columns=["bundle_size", "n_matches", "matched_countries"])`

where:
- the first column contains the number of countries supported by each bundle
- the second column contains the number of matched countries of the bundle
- the third column contains the set of matched countries of the bundle (whose length is the previous point)

Then, a subsequent for loop shall fill the rows of that dataframe.
In pseudocode:
``` python
for d_val, val in dict_sort.items():
    len_bundle = {size of the bundle}
    matched_countries = {get the matched countries using intersection and shown below}
    df_matched.loc[d_val, :] = [len_bundle, len(matched_countries), matched_countries]
```

Then, we sort the dataframe:
`df_matched.sort_values(columns=["n_matched", "bundle_size"], inplace=True, ascending=False)`

Finally, we process the dataframe as follow:
- if the first element of the sorted dataframe has n_matched == the number of desired countries, then that bundle contains all necessary data and no other bundles are needed.
   In pseudocode:
   ```
   if {first element of the dataframe has n_matched equal the number of desired countries}:
       returned_bundles = [df_matched.index[0]]
   ```
- otherwise, we need to revise the algorithm to return as many bundles as necessary to cover the bundles.
   in pseudocode:
   ```
   else:
       df_matched.drop_duplicates(subset=["matched_countries"], inplace=True)
       returned_bundles = []
       missing_countries = set(country_list)
       for d_k, dval in df_matched.items():
           {go row by row and check if the set of countries of the current row has some intersections with missing_countries, if so add d_k to returned_bundles, otherwise skip the row with continue}
    ```

The code in the for loop below can help

Reply:
Note @Emre-Yorat89 , take a screenshot of this comment because if you reset the branch it may disappear

Reply:
Okay :) Thank you.